docs/06-contributors/999-rfcs/000-stories/story-15.md

```diff
@@ -0,0 +1,186 @@
+  - [x] bring external npm package (axios)
+  - [x] bring an internal nodejs stdlib (RegEx)
+  - [x] How does untyped works with numeric operations?
+  - [x] Do we cast untyped? 
+- [x] enum & duration that can be included inside json
+- [x] It leverages setting explicit permissions (using the `this.inflight` API, described [here](https://github.com/winglang/wing/pull/1610))
+- [x] Using interface 
+- [x] Using await when calling an untyped function that returns a promise
+- [x] use redis instead of bucket
+- [ ] bring cdktf
+- [ ] code that updates estimation and duration from REST put command
+- [ ] console requirements
+
+## Discussion topics
+- `get` vs `on_get` - Should the `cloud.api` API have the `on_` prefix to match `cloud.bucket` API and also to allow calling
+the api get/post/delete/put commands (`api.get(url)` vs `api.on_get(path, inflight ())`
+- [x] Routing params - I have used express's synatx for dynamic parts of the path for the api gateway `get("/task/:id"`, 
+but then I noticed that [aws](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-api-step-by-step.html) 
+uses the `{id}` syntax. What should be the the right syntax?
+  - We will go with the [openapi spec](https://swagger.io/docs/specification/paths-and-operations/) that uses the `/tasks/{id}` syntax 
+- Bring internal nodejs types - how do we bring something that we can't require, like `RegEx`
+- From anything to Json - can I cast something from bring untyped to Json? 
+- Calling await 
+
+## Code 
+```ts (wing)
+bring cloud;
+bring redis;
+
+// TODO discuss how we bring untyped something like RegEx from JavaScript 
+// PLACEHOLER for bringing something from Javascript stdlib
+bring untyped js ;
+
+// prerequisite: npm install axios
+// PALCEHOLDER for bringing some external module
+bring untyped "axios" as axios; 
+
+enum Status {
+  Uncompleted,
+  Completed
+}
+
+interface ITaskList {
+  inflight get(id: str): Json;
+  inflight add(title: str): str;
+  inflight remove(id: str): void; 
+  inflight find(r: str): Array<str>;
+  inflight set_status(id: str, status: Status): str;
+  inflight set_estimation(id: str, estimation: duration): str;
+}
+
+resource TaskList implementes ITaskList {
+  _redis: redis.Redis;
+  init() {
+    this._redis = new redis.Redis();
+    this.inflights.add("get", ref: "this._redis", op: "GET" );
+    this.inflights.add("_add", ref: "this._redis", op: ["SET", "SADD"]);
+    // notice I am setting explicit permissions on this
+    this.inflights.add("add", ref: "this", op: "_add");
+    // is this the right synatx for multiple ops? 
+    this.inflights.add("set_status", ref: "this", op: ["_add", "get"]); 
+    this.inflights.add("set_estimation", ref: "this", op: ["_add", "get"]); 
+
+    // TODO add more permissions for remove, find
+  }
+
+  inflight get(id: str): Json {
+     return Json.parse(this._redis.GET(id));
+  }
+  
+  inflight _add(id: str, j: Json): str {
+    this._redis.SET(id , Json.format(j));
```

Comment:
In some other languages "format" is associated with format strings like `printf` in C, `println!` in Rust, or f-strings in Python. We could consider renaming to
```suggestion
    this._redis.SET(id , Json.to_str(j));
```

Reply:
Alternatively, to maybe make it more clear that this is effectively a cast:

```suggestion
    this._redis.SET(id , Json.as_str(j));
```

Reply:
I also found format confusing.

@eladb  FTI

Reply:
changed to to_str